### ISSUE TITLE
[ unused0 ] in the generated captions.

### ISSUE THREAD

**Issue by christopherGuan** [microsoft/GenerativeImage2Text#50]
By using GIT_BASE_COCO model, I expect to generate a caption: Newt Gingrich sitting in a chair.
But the generated caption is: [ unused0 ] sitting in a chair.

The image is in below.
![28](https://user-images.githubusercontent.com/45466031/233903751-1b45ab18-5801-4193-b12a-0eca7824d473.jpg)

Is there any legal issues so you cannot show a celebrity's name in a caption? In a similar issus, you mentioned that [ unused0 ]  can be replaced by <person> in GIT_BASE_COCO, but why only <person> instead that person's real name?

Also, can I train more images for the base model that eventually it can generate a caption with the celebrity's name?

The reason that I want the celebrity's real name is because I am trying to build up a image recommendation system, and the name is a word that contains a very heavy weight to recommend a accurate image for the user. 

Please explain this issue more. Thanks.

**Comment by amsword** [comment]
1. can you double check if it is GIT_BASE_COCO or GIT_LARGE_COCO? Intuitively, it should never predict [unused0] with GIT_BASE, but it may be the case for GIT_LARGE. If it is GIT_LARGE, try the model of GIT_LARGE_R.
2. the reason is that in GIT_LARGE, the pretraining dataset contains cc12m. The released dataset masks the person's name and thus it is released with a special name [unused0]. checkout the following in the readme
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/3356680/234417581-3339065a-f200-4523-aa4d-4260b5d35bd6.png">

3. To enable the model to output the person's name, you can try to fine-tune the model with more data, which contains the person's name, also as you suggested.

**Comment by christopherGuan** [comment]
Thank you so much for your response amsword.

The model i used was GIT_LARGE_TEXTCAPS. I will try to use GIT_LARGE_R_TEXTCAPS instead.

Regarding the issue of persons' names, is there any dataset that contains many person's names so that the generated caption can articulate more celebrities' names? 

For example, CelebA dataset? The dataset only contain celebrities' face images and names instead of a relatively long captions. In this case, if I train the model by using the code provided below, I have a few questions.

The code can be plugged into any trainer easily. Here is the example for the base model.
Pretraining/captioning
python -m generativeimage2text.train -p "{'type': 'forward_backward_example', \
                'image_files': ['aux_data/images/1.jpg', 'aux_data/images/2.jpg'], \
                'captions': ['a couple of boats in a large body of water.', 'a view of a mountain with a tree'], \
            }"
        
1. How to train other models like GIT_LARGE_R_TEXTCAPS instead of just the base model?
2. How to fine-tune the model? Or would you mind to train a new model that targeting celebrity name issue?
3. For example, I want the model to generate caption "Newt Gingrich sitting in a chair" when I input the image on the top, if I input 1k Newt Gingrich's image, and the input capiton is only his name ['Newt Gingrich'] for 1k images, can I eventually got a good result? 
4. After training the model, how to keep the model so that next time I can use it immediately?

**Comment by amsword** [comment]
1. yes. this might require to modify the code a bit. It should be easy.
2. as instructed, you need to wrap the code with some trainer and your dataset. Standard data loader is good enough
3. I don't think. But this can serve as pretraining, and eventually, you need to build your dataset to satisfy your requirements, either from the web crawling or from manual annotation.
4. it depends on the trainer you use and you are familiar with. Any kind of trainer normally have some sample code to play with.
